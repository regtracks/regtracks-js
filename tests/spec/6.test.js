const Track = require('../../index');

describe('tests section 6 of the specification', () => {
  test('section 6.1: charsets', () => {
    let schema = String.raw`
    @set
    a, b, (cheese), f to z, digit, whitespace
    `;

    const track = new Track(schema);
    expect(track.parser.tree.tree).toMatchSnapshot();

    let schema2 = schema + String.raw`
      @matching
      ||
      any set
      ||
    `;

    let good = 'b';
    let good2 = 'cheese';
    let good3 = 'x';
    let bad = 'e';
    let bad2 = '!';

    const track2 = new Track(schema2);
    expect(track2.test(good, 'matching')).toBe(true);
    expect(track2.test(good2, 'matching')).toBe(true);
    expect(track2.test(good3, 'matching')).toBe(true);
    expect(track2.test(bad, 'matching')).toBe(false);
    expect(track2.test(bad2, 'matching')).toBe(false);
  });

  test('section 6.2: no recursion', () => {
    let illFormedSchema = String.raw`
      @foobar
      a, b, A to Z, foobar, digit
    `;

    expect(() => new Track(illFormedSchema)).toThrow();
  });
});
