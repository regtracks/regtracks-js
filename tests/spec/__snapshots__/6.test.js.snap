// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests section 6 of the specification section 6.1: charsets 1`] = `
Object {
  "set": Object {
    "matches": Array [
      Object {
        "matches": "a",
        "type": "literal_criterion",
      },
      Object {
        "matches": "b",
        "type": "literal_criterion",
      },
      Object {
        "matches": "cheese",
        "type": "literal_criterion",
      },
      Object {
        "matches": /\\[f-z\\]/,
        "type": "character_range_criterion",
      },
      Object {
        "matches": "digit",
        "type": "symbol_criterion",
      },
      Object {
        "matches": "whitespace",
        "type": "symbol_criterion",
      },
    ],
    "name": "set",
    "type": "charset",
  },
}
`;
